hugegraph-core/src/main/java/com/baidu/hugegraph/backend/tx/GraphTransaction.java

```diff
@@ -992,15 +994,26 @@ private Query optimizeQuery(ConditionQuery query) {
                     PropertyKey pk = this.graph().propertyKey((Id) r.key());
                     r.serialValue(pk.serialValue(r.value()));
                 }
-                // Convert to sysprop condition {SORT_VALUES=value}
-                if (query.hasRangeCondition()) {
+                PropertyKey lastProp = this.graph().propertyKey(
+                                       keys.get(matchedNum - 1));
+                boolean hasRange = lastProp.dataType().isNumber() ||
+                                   lastProp.dataType().isDate();
+                List<Id> joinedKeys = keys.subList(0, matchedNum);
+                if (hasRange) {
```

Comment:
please use query.hasRangeCondition(), can't conclude that it is a range query if there are numbers in conditions